test/rulesets/LinearAlgebra/structured.jl

```diff
@@ -156,4 +156,20 @@
             rrule_test(Op, randn(n, n), (randn(n, n), randn(n, n)), (k, nothing))
         end
     end
+
+    @testset "det and logdet $T" for T in (Diagonal, UpperTriangular, LowerTriangular)
+        n = 5
+        # rand (not randn) so det will be postive, so logdet will be defined
+        X = T(rand(n, n))
```

Comment:
You might want to add a small +ve constant to these to ensure stability under changing seed -- finite differencing might make one of the values less than zero if one of the sampled values happens to be really close to zero.

Reply:
```suggestion
        X = T(3*rand(n, n) .+ 1)
```